// conekta/dist/api/shippings-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Conekta API
 * Conekta sdk
 *
 * The version of the OpenAPI document: 2.1.0
 * Contact: engineering@conekta.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrdersUpdateShippingAcceptLanguageEnum = exports.OrdersDeleteShippingAcceptLanguageEnum = exports.OrdersCreateShippingAcceptLanguageEnum = exports.ShippingsApi = exports.ShippingsApiFactory = exports.ShippingsApiFp = exports.ShippingsApiAxiosParamCreator = void 0;
const axios_1 = require("axios");
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * ShippingsApi - axios parameter creator
 * @export
 */
const ShippingsApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Create new shipping for an existing orden
         * @summary Create Shipping
         * @param {string} id Identifier of the resource
         * @param {ShippingRequest} shippingRequest requested field for a shipping
         * @param {OrdersCreateShippingAcceptLanguageEnum} [acceptLanguage] Use for knowing which language to use
         * @param {string} [xChildCompanyId] In the case of a holding company, the company id of the child company to which will process the request.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        ordersCreateShipping: (id_1, shippingRequest_1, acceptLanguage_1, xChildCompanyId_1, ...args_1) => __awaiter(this, [id_1, shippingRequest_1, acceptLanguage_1, xChildCompanyId_1, ...args_1], void 0, function* (id, shippingRequest, acceptLanguage, xChildCompanyId, options = {}) {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('ordersCreateShipping', 'id', id);
            // verify required parameter 'shippingRequest' is not null or undefined
            (0, common_1.assertParamExists)('ordersCreateShipping', 'shippingRequest', shippingRequest);
            const localVarPath = `/orders/{id}/shipping_lines`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'POST' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearerAuth required
            // http bearer authentication required
            yield (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            if (acceptLanguage != null) {
                localVarHeaderParameter['Accept-Language'] = String(acceptLanguage);
            }
            if (xChildCompanyId != null) {
                localVarHeaderParameter['X-Child-Company-Id'] = String(xChildCompanyId);
            }
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(shippingRequest, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         * Delete shipping
         * @summary Delete Shipping
         * @param {string} id Identifier of the resource
         * @param {string} shippingId identifier
         * @param {OrdersDeleteShippingAcceptLanguageEnum} [acceptLanguage] Use for knowing which language to use
         * @param {string} [xChildCompanyId] In the case of a holding company, the company id of the child company to which will process the request.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        ordersDeleteShipping: (id_2, shippingId_1, acceptLanguage_2, xChildCompanyId_2, ...args_2) => __awaiter(this, [id_2, shippingId_1, acceptLanguage_2, xChildCompanyId_2, ...args_2], void 0, function* (id, shippingId, acceptLanguage, xChildCompanyId, options = {}) {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('ordersDeleteShipping', 'id', id);
            // verify required parameter 'shippingId' is not null or undefined
            (0, common_1.assertParamExists)('ordersDeleteShipping', 'shippingId', shippingId);
            const localVarPath = `/orders/{id}/shipping_lines/{shipping_id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)))
                .replace(`{${"shipping_id"}}`, encodeURIComponent(String(shippingId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'DELETE' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearerAuth required
            // http bearer authentication required
            yield (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            if (acceptLanguage != null) {
                localVarHeaderParameter['Accept-Language'] = String(acceptLanguage);
            }
            if (xChildCompanyId != null) {
                localVarHeaderParameter['X-Child-Company-Id'] = String(xChildCompanyId);
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         * Update existing shipping for an existing orden
         * @summary Update Shipping
         * @param {string} id Identifier of the resource
         * @param {string} shippingId identifier
         * @param {ShippingRequest} shippingRequest requested field for a shipping
         * @param {OrdersUpdateShippingAcceptLanguageEnum} [acceptLanguage] Use for knowing which language to use
         * @param {string} [xChildCompanyId] In the case of a holding company, the company id of the child company to which will process the request.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        ordersUpdateShipping: (id_3, shippingId_2, shippingRequest_2, acceptLanguage_3, xChildCompanyId_3, ...args_3) => __awaiter(this, [id_3, shippingId_2, shippingRequest_2, acceptLanguage_3, xChildCompanyId_3, ...args_3], void 0, function* (id, shippingId, shippingRequest, acceptLanguage, xChildCompanyId, options = {}) {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('ordersUpdateShipping', 'id', id);
            // verify required parameter 'shippingId' is not null or undefined
            (0, common_1.assertParamExists)('ordersUpdateShipping', 'shippingId', shippingId);
            // verify required parameter 'shippingRequest' is not null or undefined
            (0, common_1.assertParamExists)('ordersUpdateShipping', 'shippingRequest', shippingRequest);
            const localVarPath = `/orders/{id}/shipping_lines/{shipping_id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)))
                .replace(`{${"shipping_id"}}`, encodeURIComponent(String(shippingId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'PUT' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearerAuth required
            // http bearer authentication required
            yield (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            if (acceptLanguage != null) {
                localVarHeaderParameter['Accept-Language'] = String(acceptLanguage);
            }
            if (xChildCompanyId != null) {
                localVarHeaderParameter['X-Child-Company-Id'] = String(xChildCompanyId);
            }
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(shippingRequest, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
    };
};
exports.ShippingsApiAxiosParamCreator = ShippingsApiAxiosParamCreator;
/**
 * ShippingsApi - functional programming interface
 * @export
 */
const ShippingsApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.ShippingsApiAxiosParamCreator)(configuration);
    return {
        /**
         * Create new shipping for an existing orden
         * @summary Create Shipping
         * @param {string} id Identifier of the resource
         * @param {ShippingRequest} shippingRequest requested field for a shipping
         * @param {OrdersCreateShippingAcceptLanguageEnum} [acceptLanguage] Use for knowing which language to use
         * @param {string} [xChildCompanyId] In the case of a holding company, the company id of the child company to which will process the request.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        ordersCreateShipping(id, shippingRequest, acceptLanguage, xChildCompanyId, options) {
            return __awaiter(this, void 0, void 0, function* () {
                var _a, _b, _c;
                const localVarAxiosArgs = yield localVarAxiosParamCreator.ordersCreateShipping(id, shippingRequest, acceptLanguage, xChildCompanyId, options);
                const index = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const operationBasePath = (_c = (_b = base_1.operationServerMap['ShippingsApi.ordersCreateShipping']) === null || _b === void 0 ? void 0 : _b[index]) === null || _c === void 0 ? void 0 : _c.url;
                return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, operationBasePath || basePath);
            });
        },
        /**
         * Delete shipping
         * @summary Delete Shipping
         * @param {string} id Identifier of the resource
         * @param {string} shippingId identifier
         * @param {OrdersDeleteShippingAcceptLanguageEnum} [acceptLanguage] Use for knowing which language to use
         * @param {string} [xChildCompanyId] In the case of a holding company, the company id of the child company to which will process the request.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        ordersDeleteShipping(id, shippingId, acceptLanguage, xChildCompanyId, options) {
            return __awaiter(this, void 0, void 0, function* () {
                var _a, _b, _c;
                const localVarAxiosArgs = yield localVarAxiosParamCreator.ordersDeleteShipping(id, shippingId, acceptLanguage, xChildCompanyId, options);
                const index = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const operationBasePath = (_c = (_b = base_1.operationServerMap['ShippingsApi.ordersDeleteShipping']) === null || _b === void 0 ? void 0 : _b[index]) === null || _c === void 0 ? void 0 : _c.url;
                return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, operationBasePath || basePath);
            });
        },
        /**
         * Update existing shipping for an existing orden
         * @summary Update Shipping
         * @param {string} id Identifier of the resource
         * @param {string} shippingId identifier
         * @param {ShippingRequest} shippingRequest requested field for a shipping
         * @param {OrdersUpdateShippingAcceptLanguageEnum} [acceptLanguage] Use for knowing which language to use
         * @param {string} [xChildCompanyId] In the case of a holding company, the company id of the child company to which will process the request.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        ordersUpdateShipping(id, shippingId, shippingRequest, acceptLanguage, xChildCompanyId, options) {
            return __awaiter(this, void 0, void 0, function* () {
                var _a, _b, _c;
                const localVarAxiosArgs = yield localVarAxiosParamCreator.ordersUpdateShipping(id, shippingId, shippingRequest, acceptLanguage, xChildCompanyId, options);
                const index = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const operationBasePath = (_c = (_b = base_1.operationServerMap['ShippingsApi.ordersUpdateShipping']) === null || _b === void 0 ? void 0 : _b[index]) === null || _c === void 0 ? void 0 : _c.url;
                return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, operationBasePath || basePath);
            });
        },
    };
};
exports.ShippingsApiFp = ShippingsApiFp;
/**
 * ShippingsApi - factory interface
 * @export
 */
const ShippingsApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.ShippingsApiFp)(configuration);
    return {
        /**
         * Create new shipping for an existing orden
         * @summary Create Shipping
         * @param {string} id Identifier of the resource
         * @param {ShippingRequest} shippingRequest requested field for a shipping
         * @param {OrdersCreateShippingAcceptLanguageEnum} [acceptLanguage] Use for knowing which language to use
         * @param {string} [xChildCompanyId] In the case of a holding company, the company id of the child company to which will process the request.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        ordersCreateShipping(id, shippingRequest, acceptLanguage, xChildCompanyId, options) {
            return localVarFp.ordersCreateShipping(id, shippingRequest, acceptLanguage, xChildCompanyId, options).then((request) => request(axios, basePath));
        },
        /**
         * Delete shipping
         * @summary Delete Shipping
         * @param {string} id Identifier of the resource
         * @param {string} shippingId identifier
         * @param {OrdersDeleteShippingAcceptLanguageEnum} [acceptLanguage] Use for knowing which language to use
         * @param {string} [xChildCompanyId] In the case of a holding company, the company id of the child company to which will process the request.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        ordersDeleteShipping(id, shippingId, acceptLanguage, xChildCompanyId, options) {
            return localVarFp.ordersDeleteShipping(id, shippingId, acceptLanguage, xChildCompanyId, options).then((request) => request(axios, basePath));
        },
        /**
         * Update existing shipping for an existing orden
         * @summary Update Shipping
         * @param {string} id Identifier of the resource
         * @param {string} shippingId identifier
         * @param {ShippingRequest} shippingRequest requested field for a shipping
         * @param {OrdersUpdateShippingAcceptLanguageEnum} [acceptLanguage] Use for knowing which language to use
         * @param {string} [xChildCompanyId] In the case of a holding company, the company id of the child company to which will process the request.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        ordersUpdateShipping(id, shippingId, shippingRequest, acceptLanguage, xChildCompanyId, options) {
            return localVarFp.ordersUpdateShipping(id, shippingId, shippingRequest, acceptLanguage, xChildCompanyId, options).then((request) => request(axios, basePath));
        },
    };
};
exports.ShippingsApiFactory = ShippingsApiFactory;
/**
 * ShippingsApi - object-oriented interface
 * @export
 * @class ShippingsApi
 * @extends {BaseAPI}
 */
class ShippingsApi extends base_1.BaseAPI {
    /**
     * Create new shipping for an existing orden
     * @summary Create Shipping
     * @param {string} id Identifier of the resource
     * @param {ShippingRequest} shippingRequest requested field for a shipping
     * @param {OrdersCreateShippingAcceptLanguageEnum} [acceptLanguage] Use for knowing which language to use
     * @param {string} [xChildCompanyId] In the case of a holding company, the company id of the child company to which will process the request.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ShippingsApi
     */
    ordersCreateShipping(id, shippingRequest, acceptLanguage, xChildCompanyId, options) {
        return (0, exports.ShippingsApiFp)(this.configuration).ordersCreateShipping(id, shippingRequest, acceptLanguage, xChildCompanyId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Delete shipping
     * @summary Delete Shipping
     * @param {string} id Identifier of the resource
     * @param {string} shippingId identifier
     * @param {OrdersDeleteShippingAcceptLanguageEnum} [acceptLanguage] Use for knowing which language to use
     * @param {string} [xChildCompanyId] In the case of a holding company, the company id of the child company to which will process the request.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ShippingsApi
     */
    ordersDeleteShipping(id, shippingId, acceptLanguage, xChildCompanyId, options) {
        return (0, exports.ShippingsApiFp)(this.configuration).ordersDeleteShipping(id, shippingId, acceptLanguage, xChildCompanyId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Update existing shipping for an existing orden
     * @summary Update Shipping
     * @param {string} id Identifier of the resource
     * @param {string} shippingId identifier
     * @param {ShippingRequest} shippingRequest requested field for a shipping
     * @param {OrdersUpdateShippingAcceptLanguageEnum} [acceptLanguage] Use for knowing which language to use
     * @param {string} [xChildCompanyId] In the case of a holding company, the company id of the child company to which will process the request.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ShippingsApi
     */
    ordersUpdateShipping(id, shippingId, shippingRequest, acceptLanguage, xChildCompanyId, options) {
        return (0, exports.ShippingsApiFp)(this.configuration).ordersUpdateShipping(id, shippingId, shippingRequest, acceptLanguage, xChildCompanyId, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.ShippingsApi = ShippingsApi;
/**
 * @export
 */
exports.OrdersCreateShippingAcceptLanguageEnum = {
    es: 'es',
    en: 'en'
};
/**
 * @export
 */
exports.OrdersDeleteShippingAcceptLanguageEnum = {
    es: 'es',
    en: 'en'
};
/**
 * @export
 */
exports.OrdersUpdateShippingAcceptLanguageEnum = {
    es: 'es',
    en: 'en'
};
