// conekta/dist/esm/api/antifraud-api.js
/* tslint:disable */
/* eslint-disable */
/**
 * Conekta API
 * Conekta sdk
 *
 * The version of the OpenAPI document: 2.1.0
 * Contact: engineering@conekta.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
import globalAxios from 'axios';
// Some imports not used depending on template conditions
// @ts-ignore
import { DUMMY_BASE_URL, assertParamExists, setBearerAuthToObject, setSearchParams, serializeDataIfNeeded, toPathString, createRequestFunction } from '../common';
// @ts-ignore
import { BASE_PATH, BaseAPI, operationServerMap } from '../base';
/**
 * AntifraudApi - axios parameter creator
 * @export
 */
export const AntifraudApiAxiosParamCreator = function (configuration) {
    return {
        /**
         *
         * @summary Create blacklisted rule
         * @param {CreateRiskRulesData} createRiskRulesData requested field for blacklist rule
         * @param {CreateRuleBlacklistAcceptLanguageEnum} [acceptLanguage] Use for knowing which language to use
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createRuleBlacklist: (createRiskRulesData_1, acceptLanguage_1, ...args_1) => __awaiter(this, [createRiskRulesData_1, acceptLanguage_1, ...args_1], void 0, function* (createRiskRulesData, acceptLanguage, options = {}) {
            // verify required parameter 'createRiskRulesData' is not null or undefined
            assertParamExists('createRuleBlacklist', 'createRiskRulesData', createRiskRulesData);
            const localVarPath = `/antifraud/blacklists`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'POST' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearerAuth required
            // http bearer authentication required
            yield setBearerAuthToObject(localVarHeaderParameter, configuration);
            if (acceptLanguage != null) {
                localVarHeaderParameter['Accept-Language'] = String(acceptLanguage);
            }
            localVarHeaderParameter['Content-Type'] = 'application/json';
            setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = serializeDataIfNeeded(createRiskRulesData, localVarRequestOptions, configuration);
            return {
                url: toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *
         * @summary Create whitelisted rule
         * @param {CreateRuleWhitelistAcceptLanguageEnum} [acceptLanguage] Use for knowing which language to use
         * @param {CreateRiskRulesData} [createRiskRulesData]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createRuleWhitelist: (acceptLanguage_2, createRiskRulesData_2, ...args_2) => __awaiter(this, [acceptLanguage_2, createRiskRulesData_2, ...args_2], void 0, function* (acceptLanguage, createRiskRulesData, options = {}) {
            const localVarPath = `/antifraud/whitelists`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'POST' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearerAuth required
            // http bearer authentication required
            yield setBearerAuthToObject(localVarHeaderParameter, configuration);
            if (acceptLanguage != null) {
                localVarHeaderParameter['Accept-Language'] = String(acceptLanguage);
            }
            localVarHeaderParameter['Content-Type'] = 'application/json';
            setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = serializeDataIfNeeded(createRiskRulesData, localVarRequestOptions, configuration);
            return {
                url: toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *
         * @summary Delete blacklisted rule
         * @param {string} id Identifier of the resource
         * @param {DeleteRuleBlacklistAcceptLanguageEnum} [acceptLanguage] Use for knowing which language to use
         * @param {string} [xChildCompanyId] In the case of a holding company, the company id of the child company to which will process the request.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteRuleBlacklist: (id_1, acceptLanguage_3, xChildCompanyId_1, ...args_3) => __awaiter(this, [id_1, acceptLanguage_3, xChildCompanyId_1, ...args_3], void 0, function* (id, acceptLanguage, xChildCompanyId, options = {}) {
            // verify required parameter 'id' is not null or undefined
            assertParamExists('deleteRuleBlacklist', 'id', id);
            const localVarPath = `/antifraud/blacklists/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'DELETE' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearerAuth required
            // http bearer authentication required
            yield setBearerAuthToObject(localVarHeaderParameter, configuration);
            if (acceptLanguage != null) {
                localVarHeaderParameter['Accept-Language'] = String(acceptLanguage);
            }
            if (xChildCompanyId != null) {
                localVarHeaderParameter['X-Child-Company-Id'] = String(xChildCompanyId);
            }
            setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *
         * @summary Delete whitelisted rule
         * @param {string} id Identifier of the resource
         * @param {DeleteRuleWhitelistAcceptLanguageEnum} [acceptLanguage] Use for knowing which language to use
         * @param {string} [xChildCompanyId] In the case of a holding company, the company id of the child company to which will process the request.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteRuleWhitelist: (id_2, acceptLanguage_4, xChildCompanyId_2, ...args_4) => __awaiter(this, [id_2, acceptLanguage_4, xChildCompanyId_2, ...args_4], void 0, function* (id, acceptLanguage, xChildCompanyId, options = {}) {
            // verify required parameter 'id' is not null or undefined
            assertParamExists('deleteRuleWhitelist', 'id', id);
            const localVarPath = `/antifraud/whitelists/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'DELETE' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearerAuth required
            // http bearer authentication required
            yield setBearerAuthToObject(localVarHeaderParameter, configuration);
            if (acceptLanguage != null) {
                localVarHeaderParameter['Accept-Language'] = String(acceptLanguage);
            }
            if (xChildCompanyId != null) {
                localVarHeaderParameter['X-Child-Company-Id'] = String(xChildCompanyId);
            }
            setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         * Return all rules
         * @summary Get list of blacklisted rules
         * @param {GetRuleBlacklistAcceptLanguageEnum} [acceptLanguage] Use for knowing which language to use
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getRuleBlacklist: (acceptLanguage_5, ...args_5) => __awaiter(this, [acceptLanguage_5, ...args_5], void 0, function* (acceptLanguage, options = {}) {
            const localVarPath = `/antifraud/blacklists`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearerAuth required
            // http bearer authentication required
            yield setBearerAuthToObject(localVarHeaderParameter, configuration);
            if (acceptLanguage != null) {
                localVarHeaderParameter['Accept-Language'] = String(acceptLanguage);
            }
            setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         * Return all rules
         * @summary Get a list of whitelisted rules
         * @param {GetRuleWhitelistAcceptLanguageEnum} [acceptLanguage] Use for knowing which language to use
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getRuleWhitelist: (acceptLanguage_6, ...args_6) => __awaiter(this, [acceptLanguage_6, ...args_6], void 0, function* (acceptLanguage, options = {}) {
            const localVarPath = `/antifraud/whitelists`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearerAuth required
            // http bearer authentication required
            yield setBearerAuthToObject(localVarHeaderParameter, configuration);
            if (acceptLanguage != null) {
                localVarHeaderParameter['Accept-Language'] = String(acceptLanguage);
            }
            setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
    };
};
/**
 * AntifraudApi - functional programming interface
 * @export
 */
export const AntifraudApiFp = function (configuration) {
    const localVarAxiosParamCreator = AntifraudApiAxiosParamCreator(configuration);
    return {
        /**
         *
         * @summary Create blacklisted rule
         * @param {CreateRiskRulesData} createRiskRulesData requested field for blacklist rule
         * @param {CreateRuleBlacklistAcceptLanguageEnum} [acceptLanguage] Use for knowing which language to use
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createRuleBlacklist(createRiskRulesData, acceptLanguage, options) {
            return __awaiter(this, void 0, void 0, function* () {
                var _a, _b, _c;
                const localVarAxiosArgs = yield localVarAxiosParamCreator.createRuleBlacklist(createRiskRulesData, acceptLanguage, options);
                const index = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const operationBasePath = (_c = (_b = operationServerMap['AntifraudApi.createRuleBlacklist']) === null || _b === void 0 ? void 0 : _b[index]) === null || _c === void 0 ? void 0 : _c.url;
                return (axios, basePath) => createRequestFunction(localVarAxiosArgs, globalAxios, BASE_PATH, configuration)(axios, operationBasePath || basePath);
            });
        },
        /**
         *
         * @summary Create whitelisted rule
         * @param {CreateRuleWhitelistAcceptLanguageEnum} [acceptLanguage] Use for knowing which language to use
         * @param {CreateRiskRulesData} [createRiskRulesData]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createRuleWhitelist(acceptLanguage, createRiskRulesData, options) {
            return __awaiter(this, void 0, void 0, function* () {
                var _a, _b, _c;
                const localVarAxiosArgs = yield localVarAxiosParamCreator.createRuleWhitelist(acceptLanguage, createRiskRulesData, options);
                const index = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const operationBasePath = (_c = (_b = operationServerMap['AntifraudApi.createRuleWhitelist']) === null || _b === void 0 ? void 0 : _b[index]) === null || _c === void 0 ? void 0 : _c.url;
                return (axios, basePath) => createRequestFunction(localVarAxiosArgs, globalAxios, BASE_PATH, configuration)(axios, operationBasePath || basePath);
            });
        },
        /**
         *
         * @summary Delete blacklisted rule
         * @param {string} id Identifier of the resource
         * @param {DeleteRuleBlacklistAcceptLanguageEnum} [acceptLanguage] Use for knowing which language to use
         * @param {string} [xChildCompanyId] In the case of a holding company, the company id of the child company to which will process the request.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteRuleBlacklist(id, acceptLanguage, xChildCompanyId, options) {
            return __awaiter(this, void 0, void 0, function* () {
                var _a, _b, _c;
                const localVarAxiosArgs = yield localVarAxiosParamCreator.deleteRuleBlacklist(id, acceptLanguage, xChildCompanyId, options);
                const index = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const operationBasePath = (_c = (_b = operationServerMap['AntifraudApi.deleteRuleBlacklist']) === null || _b === void 0 ? void 0 : _b[index]) === null || _c === void 0 ? void 0 : _c.url;
                return (axios, basePath) => createRequestFunction(localVarAxiosArgs, globalAxios, BASE_PATH, configuration)(axios, operationBasePath || basePath);
            });
        },
        /**
         *
         * @summary Delete whitelisted rule
         * @param {string} id Identifier of the resource
         * @param {DeleteRuleWhitelistAcceptLanguageEnum} [acceptLanguage] Use for knowing which language to use
         * @param {string} [xChildCompanyId] In the case of a holding company, the company id of the child company to which will process the request.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteRuleWhitelist(id, acceptLanguage, xChildCompanyId, options) {
            return __awaiter(this, void 0, void 0, function* () {
                var _a, _b, _c;
                const localVarAxiosArgs = yield localVarAxiosParamCreator.deleteRuleWhitelist(id, acceptLanguage, xChildCompanyId, options);
                const index = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const operationBasePath = (_c = (_b = operationServerMap['AntifraudApi.deleteRuleWhitelist']) === null || _b === void 0 ? void 0 : _b[index]) === null || _c === void 0 ? void 0 : _c.url;
                return (axios, basePath) => createRequestFunction(localVarAxiosArgs, globalAxios, BASE_PATH, configuration)(axios, operationBasePath || basePath);
            });
        },
        /**
         * Return all rules
         * @summary Get list of blacklisted rules
         * @param {GetRuleBlacklistAcceptLanguageEnum} [acceptLanguage] Use for knowing which language to use
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getRuleBlacklist(acceptLanguage, options) {
            return __awaiter(this, void 0, void 0, function* () {
                var _a, _b, _c;
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getRuleBlacklist(acceptLanguage, options);
                const index = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const operationBasePath = (_c = (_b = operationServerMap['AntifraudApi.getRuleBlacklist']) === null || _b === void 0 ? void 0 : _b[index]) === null || _c === void 0 ? void 0 : _c.url;
                return (axios, basePath) => createRequestFunction(localVarAxiosArgs, globalAxios, BASE_PATH, configuration)(axios, operationBasePath || basePath);
            });
        },
        /**
         * Return all rules
         * @summary Get a list of whitelisted rules
         * @param {GetRuleWhitelistAcceptLanguageEnum} [acceptLanguage] Use for knowing which language to use
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getRuleWhitelist(acceptLanguage, options) {
            return __awaiter(this, void 0, void 0, function* () {
                var _a, _b, _c;
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getRuleWhitelist(acceptLanguage, options);
                const index = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const operationBasePath = (_c = (_b = operationServerMap['AntifraudApi.getRuleWhitelist']) === null || _b === void 0 ? void 0 : _b[index]) === null || _c === void 0 ? void 0 : _c.url;
                return (axios, basePath) => createRequestFunction(localVarAxiosArgs, globalAxios, BASE_PATH, configuration)(axios, operationBasePath || basePath);
            });
        },
    };
};
/**
 * AntifraudApi - factory interface
 * @export
 */
export const AntifraudApiFactory = function (configuration, basePath, axios) {
    const localVarFp = AntifraudApiFp(configuration);
    return {
        /**
         *
         * @summary Create blacklisted rule
         * @param {CreateRiskRulesData} createRiskRulesData requested field for blacklist rule
         * @param {CreateRuleBlacklistAcceptLanguageEnum} [acceptLanguage] Use for knowing which language to use
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createRuleBlacklist(createRiskRulesData, acceptLanguage, options) {
            return localVarFp.createRuleBlacklist(createRiskRulesData, acceptLanguage, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Create whitelisted rule
         * @param {CreateRuleWhitelistAcceptLanguageEnum} [acceptLanguage] Use for knowing which language to use
         * @param {CreateRiskRulesData} [createRiskRulesData]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createRuleWhitelist(acceptLanguage, createRiskRulesData, options) {
            return localVarFp.createRuleWhitelist(acceptLanguage, createRiskRulesData, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Delete blacklisted rule
         * @param {string} id Identifier of the resource
         * @param {DeleteRuleBlacklistAcceptLanguageEnum} [acceptLanguage] Use for knowing which language to use
         * @param {string} [xChildCompanyId] In the case of a holding company, the company id of the child company to which will process the request.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteRuleBlacklist(id, acceptLanguage, xChildCompanyId, options) {
            return localVarFp.deleteRuleBlacklist(id, acceptLanguage, xChildCompanyId, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Delete whitelisted rule
         * @param {string} id Identifier of the resource
         * @param {DeleteRuleWhitelistAcceptLanguageEnum} [acceptLanguage] Use for knowing which language to use
         * @param {string} [xChildCompanyId] In the case of a holding company, the company id of the child company to which will process the request.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteRuleWhitelist(id, acceptLanguage, xChildCompanyId, options) {
            return localVarFp.deleteRuleWhitelist(id, acceptLanguage, xChildCompanyId, options).then((request) => request(axios, basePath));
        },
        /**
         * Return all rules
         * @summary Get list of blacklisted rules
         * @param {GetRuleBlacklistAcceptLanguageEnum} [acceptLanguage] Use for knowing which language to use
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getRuleBlacklist(acceptLanguage, options) {
            return localVarFp.getRuleBlacklist(acceptLanguage, options).then((request) => request(axios, basePath));
        },
        /**
         * Return all rules
         * @summary Get a list of whitelisted rules
         * @param {GetRuleWhitelistAcceptLanguageEnum} [acceptLanguage] Use for knowing which language to use
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getRuleWhitelist(acceptLanguage, options) {
            return localVarFp.getRuleWhitelist(acceptLanguage, options).then((request) => request(axios, basePath));
        },
    };
};
/**
 * AntifraudApi - object-oriented interface
 * @export
 * @class AntifraudApi
 * @extends {BaseAPI}
 */
export class AntifraudApi extends BaseAPI {
    /**
     *
     * @summary Create blacklisted rule
     * @param {CreateRiskRulesData} createRiskRulesData requested field for blacklist rule
     * @param {CreateRuleBlacklistAcceptLanguageEnum} [acceptLanguage] Use for knowing which language to use
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof AntifraudApi
     */
    createRuleBlacklist(createRiskRulesData, acceptLanguage, options) {
        return AntifraudApiFp(this.configuration).createRuleBlacklist(createRiskRulesData, acceptLanguage, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Create whitelisted rule
     * @param {CreateRuleWhitelistAcceptLanguageEnum} [acceptLanguage] Use for knowing which language to use
     * @param {CreateRiskRulesData} [createRiskRulesData]
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof AntifraudApi
     */
    createRuleWhitelist(acceptLanguage, createRiskRulesData, options) {
        return AntifraudApiFp(this.configuration).createRuleWhitelist(acceptLanguage, createRiskRulesData, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Delete blacklisted rule
     * @param {string} id Identifier of the resource
     * @param {DeleteRuleBlacklistAcceptLanguageEnum} [acceptLanguage] Use for knowing which language to use
     * @param {string} [xChildCompanyId] In the case of a holding company, the company id of the child company to which will process the request.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof AntifraudApi
     */
    deleteRuleBlacklist(id, acceptLanguage, xChildCompanyId, options) {
        return AntifraudApiFp(this.configuration).deleteRuleBlacklist(id, acceptLanguage, xChildCompanyId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Delete whitelisted rule
     * @param {string} id Identifier of the resource
     * @param {DeleteRuleWhitelistAcceptLanguageEnum} [acceptLanguage] Use for knowing which language to use
     * @param {string} [xChildCompanyId] In the case of a holding company, the company id of the child company to which will process the request.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof AntifraudApi
     */
    deleteRuleWhitelist(id, acceptLanguage, xChildCompanyId, options) {
        return AntifraudApiFp(this.configuration).deleteRuleWhitelist(id, acceptLanguage, xChildCompanyId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Return all rules
     * @summary Get list of blacklisted rules
     * @param {GetRuleBlacklistAcceptLanguageEnum} [acceptLanguage] Use for knowing which language to use
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof AntifraudApi
     */
    getRuleBlacklist(acceptLanguage, options) {
        return AntifraudApiFp(this.configuration).getRuleBlacklist(acceptLanguage, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Return all rules
     * @summary Get a list of whitelisted rules
     * @param {GetRuleWhitelistAcceptLanguageEnum} [acceptLanguage] Use for knowing which language to use
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof AntifraudApi
     */
    getRuleWhitelist(acceptLanguage, options) {
        return AntifraudApiFp(this.configuration).getRuleWhitelist(acceptLanguage, options).then((request) => request(this.axios, this.basePath));
    }
}
/**
 * @export
 */
export const CreateRuleBlacklistAcceptLanguageEnum = {
    es: 'es',
    en: 'en'
};
/**
 * @export
 */
export const CreateRuleWhitelistAcceptLanguageEnum = {
    es: 'es',
    en: 'en'
};
/**
 * @export
 */
export const DeleteRuleBlacklistAcceptLanguageEnum = {
    es: 'es',
    en: 'en'
};
/**
 * @export
 */
export const DeleteRuleWhitelistAcceptLanguageEnum = {
    es: 'es',
    en: 'en'
};
/**
 * @export
 */
export const GetRuleBlacklistAcceptLanguageEnum = {
    es: 'es',
    en: 'en'
};
/**
 * @export
 */
export const GetRuleWhitelistAcceptLanguageEnum = {
    es: 'es',
    en: 'en'
};
